contracts/IdleYieldSource.sol

```diff
@@ -56,7 +56,7 @@ contract IdleYieldSource is IProtocolYieldSource, Initializable, ReentrancyGuard
     function initialize(
         address _idleToken
     ) public initializer {
-
+        require(address(_idleToken) != address(0), "IdleYieldSource/idleToken-not-zero");
```

Comment:
We should also verify that the address passed is a contract. For this, we can use the util function `isContract` from OpenZeppelin, it will also check for the zero address.
https://docs.openzeppelin.com/contracts/4.x/api/utils#Address-isContract-address-

Add also some unit tests to test that we can't initialize if we pass address zero or a random address that is not a contract.

Reply:
After talking with Brendan, we decided that only checking for address zero is enough. That being said the error message could be a bit more explicit. 
```suggestion
        require(address(_idleToken) != address(0), "IdleYieldSource/idleToken-not-zero-address");
```